input/pagecontent/reference.md

```diff
@@ -1190,78 +1311,92 @@ Additional keywords are as follows:
   ```
 
-#### Defining Profiles
+#### Defining Code Systems
+It is sometimes necessary to define new codes inside an IG that are not drawn from an external code system (aka _local codes_). When defining local codes, you must define them in the context of a code system.
 
-To define a profile, the keywords `Profile` and `Parent` are required, and `Id`, `Title`, and `Description` are optional.
+> **Note:** Defining local codes is not generally recommended, since those codes will not be part of recognized terminology systems. However, when existing vocabularies do not contain necessary codes, it may be necessary to define them -- at least temporarily -- as local codes.
 
-**Example:**
+Creating a code system uses the keywords `CodeSystem`, `Id`, `Title` and `Description`. Codes are then added, one per rule, using the following syntax:
 
-* Define a profile for USCorePatient:
+
+```
+* #{code} "{display string}" "{definition string}"
+```
+
+**Notes:**
+* Do not put a code system before the hash sign `#`. The code system name is given by the `CodeSystem` keyword.
+* The definition of the term can be optionally provided as the second string following the code.
+* Do not use the word `include` in a code system rule. The rule is creating a brand new code, not including an existing code defined elsewhere.
+
+**Example:** Define a code system for yoga poses.
 
   ```
-  Profile:        USCorePatient
-  Parent:         Patient
-  Id:             us-core-patient
-  Title:          "US Core Patient Profile"
-  Description:    "Defines constraints and extensions on the patient resource for the minimal set of data to query and retrieve patient demographic information."
+  CodeSystem:  YogaCS
+  Title: "Yoga Code System."
+  Description:  "A brief vocabulary of yoga-related terms."
+  * #Sirsasana "Headstand" "An inverted asana, also called mudra in classical hatha yoga, involves standing on one's head."
+  * #Halasana "Plough Pose" "Halasana or Plough pose is an inverted asana in hatha yoga and modern yoga as exercise. Its variations include Karnapidasana with the knees by the ears, and Supta Konasana with the feet wide apart."
+  * #Matsyasana "Fish Pose"  "Matsyasana is a reclining back-bending asana in hatha yoga and modern yoga as exercise. It is commonly considered a counterasana to Sarvangasana, or shoulder stand, specifically within the context of the Ashtanga Vinyasa Yoga Primary Series."
+  * #Bhujangasana "Cobra Pose" "Bhujangasana, or Cobra Pose is a reclining back-bending asana in hatha yoga and modern yoga as exercise. It is commonly performed in a cycle of asanas in Surya Namaskar (Salute to the Sun) as an alternative to Urdhva Mukha Svanasana (Upwards Dog Pose)."
   ```
 
-Any rules defined for the profiles would follow immediately after the keyword section.
+> **Note:** FSH does not currently support definition of relationships between local codes, such as parent-child (is-a) relationships.
 
 #### Defining Extensions
 
```

Comment:
Here's what I wrote:
All extensions have the same structure, but extensions can either have a value (i.e. a value[x] element) or sub-extensions, but not both. To create a simple extension, the value[x] element should be constrained. To create a complex extension, the extension array of the extension must be sliced (see [Contains Rules for Extensions](#contains-rules-for-extensions)).

Since simple and complex extensions are mutually-exclusive, FSH implementations should set the value[x] cardinality to 0..0 if sub-extensions are specified, set extension cardinality to 0..0 if constraints are applied to value[x], and signal an error if value[x] and extensions are simultaneously specified.